src/wxflow/fsutils.py

```diff
@@ -1,34 +1,45 @@
-import contextlib
-import errno
 import grp
 import os
 import shutil
+from contextlib import contextmanager
+from logging import getLogger
 
 __all__ = ['mkdir', 'mkdir_p', 'rmdir', 'chdir', 'rm_p', 'cp',
            'get_gid', 'chgrp']
 
+logger = getLogger(__name__.split('.')[-1])
+
 
 def mkdir_p(path):
     try:
-        os.makedirs(path)
-    except OSError as exc:
-        if exc.errno == errno.EEXIST and os.path.isdir(path):
-            pass
-        else:
-            raise OSError(f"unable to create directory at {path}")
+        os.makedirs(path, exist_ok=True)
+    except OSError:
+        raise OSError(f"unable to create directory at {path}")
 
 
 mkdir = mkdir_p
 
 
-def rmdir(dir_path):
+def rmdir(dir_path, missing_ok=False):
+    """
+    Attempt to delete a directory and all of its contents.
+    If ignore_missing is True, then a missing directory will not raise an error.
+    """
+
     try:
         shutil.rmtree(dir_path)
-    except OSError as exc:
-        raise OSError(f"unable to remove {dir_path}")
 
+    except FileNotFoundError as exc:
+        if missing_ok:
+            logger.warning(f"WARNING cannot remove the target path {dir_path} because it does not exist")
+        else:
+            raise exc
```

Comment:
```suggestion
            raise FileNotFoundError()
```
Should this raise this error specifically.  Not sure how this works, if its trying to remove something that is not existant.

Reply:
Since `exc` is the original `FileNotFoundError`, raising `exc` is equivalent to raising `FileNotFoundError`.  But, raising a custom `FileNotFoundError` with an more specific error message would be useful.  I'll add that in.

So if the file doesn't exist, nothing will happen but `rmtree` will return a `FileNotFoundError`.  If `missing_ok=False`, then this Exception will be passed on.  Otherwise, you get a warning for trying.